### ISSUE TITLE
react colors are incorrect in the default config

### ISSUE THREAD

**Issue by nshen** [HiPhish/nvim-ts-rainbow2#18]
**Describe the bug**

jsx tag colors are incorrect

<img width="396" alt="image" src="https://user-images.githubusercontent.com/181506/226235589-d103ca6d-187c-423f-99a5-488ecf3d712d.png">

**Comment by HiPhish** [comment]
I know, but that's an issue with Neovim's Tree-sitter API. Let's take a look at the query:

```
(jsx_element
  (jsx_opening_element ["<" ">"] @opening)
  (jsx_closing_element ["<" "/" ">"] @closing)) @container
```

The capture group `@opening` matches two nodes (`"<"` and `">"`), but Neovim's `Query:iter_captures` function will only return the first node. The PR neovim/neovim#17099 aims to fix the issue, but it is still open. Tree-sitter itself is capable of iterating through the matched nodes, this is purely a Lua bindings issue.

As a workaround you could use the query `rainbow-tags-react` instead, it will highlight the entire tag, not just the angle brackets. I should add a note to the README about this limitation.

**Comment by binhtran432k** [comment]
I am also facing this error, but for me highlighting an entire tag or pair of "<", ">" is not as clear and beautiful as just the tag name. So I create my own query for typescriptreact and html as follows:

- tsx/rainbow-parens.scm
```query
; inherits: typescript

(jsx_element
  (jsx_opening_element
    name: (_) @opening)
  (jsx_closing_element
    name: (_) @closing)) @container

(jsx_self_closing_element
  name: (_) @intermediate) @container
```

- html/rainbow-tags.scm
```query
;;; A pair of opening and closing tag with any content in-between. Excludes
;;; self-closing tags or opening tags without closing tag.

(element
  (start_tag
    (tag_name) @opening)
  (end_tag
    (tag_name) @closing)) @container

(style_element
  (start_tag
    (tag_name) @opening)
  (end_tag
    (tag_name) @closing)) @container

(script_element
  (start_tag
    (tag_name) @opening)
  (end_tag
    (tag_name) @closing)) @container

(self_closing_tag
  (tag_name) @intermediate) @container
```

- Demo:
![image](https://user-images.githubusercontent.com/26183671/230256001-e28b56f7-8a51-416b-b33c-52aafc0064eb.png)

**Comment by MaxAntony** [comment]
> I am also facing this error, but for me highlighting an entire tag or pair of "<", ">" is not as clear and beautiful as just the tag name. So I create my own query for typescriptreact and html as follows:
> 
>     * tsx/rainbow-parens.scm
> 
> 
> ```
> ; inherits: typescript
> 
> (jsx_element
>   (jsx_opening_element
>     name: (_) @opening)
>   (jsx_closing_element
>     name: (_) @closing)) @container
> 
> (jsx_self_closing_element
>   name: (_) @intermediate) @container
> ```
> 
>     * html/rainbow-tags.scm
> 
> 
> ```
> ;;; A pair of opening and closing tag with any content in-between. Excludes
> ;;; self-closing tags or opening tags without closing tag.
> 
> (element
>   (start_tag
>     (tag_name) @opening)
>   (end_tag
>     (tag_name) @closing)) @container
> 
> (style_element
>   (start_tag
>     (tag_name) @opening)
>   (end_tag
>     (tag_name) @closing)) @container
> 
> (script_element
>   (start_tag
>     (tag_name) @opening)
>   (end_tag
>     (tag_name) @closing)) @container
> 
> (self_closing_tag
>   (tag_name) @intermediate) @container
> ```
> 
>     * Demo:
>       ![image](https://user-images.githubusercontent.com/26183671/230256001-e28b56f7-8a51-416b-b33c-52aafc0064eb.png)

this is exactly what i was looking for, thanks

**Comment by gungun974** [comment]
> I am also facing this error, but for me highlighting an entire tag or pair of "<", ">" is not as clear and beautiful as just the tag name. So I create my own query for typescriptreact and html as follows:
> 
>     * tsx/rainbow-parens.scm
> 
> 
> ```
> ; inherits: typescript
> 
> (jsx_element
>   (jsx_opening_element
>     name: (_) @opening)
>   (jsx_closing_element
>     name: (_) @closing)) @container
> 
> (jsx_self_closing_element
>   name: (_) @intermediate) @container
> ```
> 
>     * html/rainbow-tags.scm
> 
> 
> ```
> ;;; A pair of opening and closing tag with any content in-between. Excludes
> ;;; self-closing tags or opening tags without closing tag.
> 
> (element
>   (start_tag
>     (tag_name) @opening)
>   (end_tag
>     (tag_name) @closing)) @container
> 
> (style_element
>   (start_tag
>     (tag_name) @opening)
>   (end_tag
>     (tag_name) @closing)) @container
> 
> (script_element
>   (start_tag
>     (tag_name) @opening)
>   (end_tag
>     (tag_name) @closing)) @container
> 
> (self_closing_tag
>   (tag_name) @intermediate) @container
> ```
> 
>     * Demo:
>       ![image](https://user-images.githubusercontent.com/26183671/230256001-e28b56f7-8a51-416b-b33c-52aafc0064eb.png)

@binhtran432k thanks for your response because.

I just discovered this plugin and I have the same problem in my Vue 3 files where the HTML is not colored correctly.

However, could you tell me where you have placed your query files ?

I couldn't find anywhere in the documentation or the code on how to create local queries without having to modify the project's source code.

**Comment by HiPhish** [comment]
> I couldn't find anywhere in the documentation or the code on how to create local queries without having to modify the project's source code.

You place them wherever you want in a `queries` directory in your runtimepath. For example `~/.config/nvim/queries/tsx/my-own-query.scm`. See `:h 'rtp'` for information on runtime paths.

**Comment by benediktms** [comment]
> As a workaround you could use the query rainbow-tags-react instead, it will highlight the entire tag, not just the angle brackets. I should add a note to the README about this limitation.

For me this works, however it will also disable the standard bracket `([{` highlighting in `jsx`/`tsx` files.

**Comment by HiPhish** [comment]
@benediktms This should be fixed on the current `master` branch now, please try it out.

**Comment by benediktms** [comment]
yep seems to work now!

**Comment by HiPhish** [comment]
Hi everyone, I want to inform you that the default React query on the current `master` now highlights only the names of tags. This is in line with the query posted above. Highlighting the entire tag is overboard since tags in React projects can get very large. I did not know this when I originally wrote the query, all the examples I saw online were neat one-line tags. That is not what real React code in production looks like though.

I will leave this issue open for a while. If no one complains I will close it.
